projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php

```diff
@@ -375,6 +375,10 @@ public function add_options_menu() {
 			$submenus_to_update['options-writing.php'] = 'https://wordpress.com/settings/writing/' . $this->domain;
 		}
 
+		if ( apply_filters( 'calypso_use_modernized_reading_settings', false ) && self::DEFAULT_VIEW === $this->get_preferred_view( 'options-reading.php' ) ) {
```

Comment:
`self::DEFAULT_VIEW === $this->get_preferred_view( 'options-reading.php' )` returns `false` here because the preferred view for the Reading setting page is currently `classic`.

As a result, we get to the old WP Admin page every time - until the view is changed.

What we could consider is to use `set_preferred_view()` to set the view to `default`. Or maybe you will have an alternative idea. 🙂 

This also makes me think about the rollout - I think that once the new Reading page is ready to be launched, we can consider switching every user to use the `default` view.

It is not yet clear to me what would be the best approach to accomplish that. Looking at `set_preferred_view()`, it is using `update_user_option()` to set the option for specific user.

Reply:
Interesting. I did not even think about this condition much because it worked for my case for both simple and atomic. 

> Or maybe you will have an alternative idea. 🙂

Since this PR is not really about switch, I would consider even removing the part of the condition and just leaving out the flag condition.

However this does not mean we will not have to deal with this but rather I feel like this should be dealt with during implementation of the **Classic <-> Default** switch for the page.

Reply:
> [...] it worked for my case for both simple and atomic.

I wonder how come the value was different for us. 🤔 

> Since this PR is not really about switch, I would consider even removing the part of the condition and just leaving out the flag condition.

Yes, I think that makes sense. We can then introduce it back.

> However this does not mean we will not have to deal with this but rather I feel like this should be dealt with during implementation of the Classic <-> Default switch for the page.

That sounds good to me. 🙂👍🏼

Reply:
Let me know if you folks discover any way around it based on `Base_Admin_Menu::get_preferred_view`. I see by setting `fallback` to `false`, it will return `UNKNOWN_VIEW`, which is likely what we need, but ASSUMING we are getting the `CLASSIC_VIEW` value because of the fallback in place, and that the `jetpack_admin_menu_preferred_views` user option actually does not contain a set value for `options-reading.php`.